internal/cli/config.go

```diff
@@ -0,0 +1,137 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/auth0/auth0-cli/internal/auth"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+var requiredScopes = auth.RequiredScopes()
+
+var desiredInputs = `Config init is intended for non-interactive use, 
+ensure the following env variables are set: 
+
+AUTH0_CLI_CLIENT_DOMAIN 
+AUTH0_CLI_CLIENT_ID
+AUTH0_CLI_CLIENT_SECRET
+
+Interactive logins should use "auth0 login" instead.`
+
+type params struct {
+	filePath     string
+	clientDomain string
+	clientID     string
+	clientSecret string
+}
+
+func (p params) validate() error {
+	if p.clientDomain == "" {
+		return fmt.Errorf("Missing client domain.\n%s", desiredInputs)
+	}
+
+	u, err := url.Parse(p.clientDomain)
+	if err != nil {
+		return fmt.Errorf("Failed to parse client domain: %s", p.clientDomain)
+	}
+
+	if u.Scheme != "" {
+		return fmt.Errorf("Client domain cant include a scheme: %s", p.clientDomain)
+	}
+
+	if p.clientID == "" {
+		return fmt.Errorf("Missing client id.\n%s", desiredInputs)
+	}
+
+	if p.clientSecret == "" {
+		return fmt.Errorf("Missing client secret.\n%s", desiredInputs)
+	}
+	return nil
+}
+
+func configCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config",
+		Short: "Manage auth0-cli config",
+		Long:  "Manage auth0-cli config",
+	}
+
+	cmd.AddCommand(initCmd(cli))
+	return cmd
+}
+
+func initCmd(cli *cli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "initialize valid cli config from environment variables",
```

Comment:
I'd suggest
```suggestion
		Short: "configure the CLI from environment variables",
```

otherwise it is too internal